src/containers/treePage/ducks/selectors.ts

```diff
@@ -85,3 +91,34 @@ export const isTreeAdoptedByUser = (
     return false;
   }
 };
+
+export const mapStewardshipToMonthYearOptions = createSelector(
+  [mapStewardshipToTreeCare],
+  (stewardship: TreeCare[]): MonthYearOption[] => {
+    const monthYearOptions = stewardship
+      // turn the filtered array into an array of labels and values
+      .map((entry) => {
+        return {
+          month: entry.month,
+          year: entry.year,
+        };
+      })
+      // append the current month and year to the options list
+      .concat([
+        {
+          month: new Date().toLocaleString('default', { month: 'short' }),
+          year: new Date().getFullYear(),
+        },
+      ])
+      // remove entries with duplicate month and year
+      .filter(
+        (entry, index, self) =>
+          index ===
+          self.findIndex(
+            (e) => e.month === entry.month && e.year === entry.year,
+          ),
+      );
+    // sort dates by both month and year in reverse chronological order
+    return monthYearOptions.sort(compareByMonthYear).reverse();
```

Comment:
💭  You could have compareByMonthYear return reverse chronological order instead of reversing after, but the list will be small so it shouldn't matter.